Makefile.am

```diff
@@ -12,6 +12,7 @@ if ENABLE_MAN
 SUBDIRS += doc/man
 endif
 .PHONY: deploy FORCE
+.INTERMEDIATE: $(OSX_TEMP_ISO) $(COVERAGE_INFO)
```

Comment:
Is it safe/correct to have OSX_TEMP_ISO here even though it's not a thing for non-OSX? It is unconditionally assigned a value on line 38.

Reply:
> Is it safe/correct to have OSX_TEMP_ISO here even though it's not a thing for non-OSX?

"OSX_TEMP_ISO ... [i]s not a thing for non-OSX" seems like a bit vague statement for the next reasons:
1) Being a global variable, the `OSX_TEMP_ISO` is defined regardless of the place of its evaluation, i.e., on all platforms:
```
$ make print-OSX_TEMP_ISO
OSX_TEMP_ISO = Bitcoin-Core.temp.iso
```

2) On macOS, when `BUILD_DARWIN` is set to `true`, the resulted `Makefile` has no the `Bitcoin-Core.temp.iso` target. So one of prerequisites of the `.INTERMEDIATE` target has no recipe of its own. `make` is aware of such possible conditions (e.g., https://www.gnu.org/software/make/manual/html_node/Last-Resort.html)

3) Being a special target, the `.INTERMEDIATE` just explicitly [marks](https://www.gnu.org/software/make/manual/html_node/Chained-Rules.html) its each prerequisite file as intermediate, regardless if the file is mentioned explicitly in some other way

Reply:
Yes I was a bit unclear there, sorry.
I wondered if it might have been better to define `OSX_TEMP_ISO` as empty on non-DARWIN, instead of assign it a value, so that it's not included in `.INTERMEDIATES`.
But it is good to know that adding it to  `.INTERMEDIATES` cannot actually trigger the build rule. I guess that addresses my concern.